controller/src/main/java/org/jboss/as/controller/ModelControllerImpl.java

```diff
@@ -898,6 +898,11 @@ void addFailureDescription(ModelNode operation, ModelNode failure) {
 
     PathAddress getModelControllerResourceAddress(ManagementModel managementModel) {
         if (modelControllerResourceAddress == null) {
+            Set<String> hosts = managementModel.getRootResource().getChildrenNames(HOST);
+            // if we don't actually have a host name yet (e.g. --empty-host-config) we return null for now.
+            if (hosts.size() == 0) {
+                return null;
+            }
```

Comment:
Is this directly related to the JMX problem, or just something you noticed as you were working this issue? If the later please file a separate JIRA and split the commit. I can certainly see why this would need to be fixed, so no problem with the change.